src/coverlet.MTP/Configuration/CoverageConfiguration.cs

```diff
@@ -0,0 +1,273 @@
+﻿// Copyright (c) Toni Solarin-Sodara
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Coverlet.MTP.CommandLine;
+using Microsoft.Testing.Platform.CommandLine;
+using Microsoft.Testing.Platform.Logging;
+
+namespace Coverlet.MTP.Configuration;
+
+internal sealed class CoverageConfiguration
+{
+  private readonly ICommandLineOptions _commandLineOptions;
+  private readonly ILogger? _logger;
+
+  // Default exclusions matching coverlet.collector behavior
+  // See: https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/VSTestIntegration.md
+  private static readonly string[] s_defaultExcludeFilters = new[]
+  {
+    "[xunit.*]*",
+    "[Microsoft.Testing.*]*",
+    "[coverlet.*]*"
+  };
+
+  private static readonly string[] s_defaultExcludeByAttributes = new[]
+  {
+    "ExcludeFromCodeCoverage",
+    "ExcludeFromCodeCoverageAttribute",
+    "GeneratedCodeAttribute",
+    "CompilerGeneratedAttribute"
+  };
+
+  public CoverageConfiguration(ICommandLineOptions commandLineOptions, ILogger? logger = null)
+  {
+    _commandLineOptions = commandLineOptions;
+    _logger = logger;
+  }
+
+  public bool IsCoverageEnabled =>
+    _commandLineOptions.IsOptionSet(CoverletOptionNames.Coverage);
+
+  public string[] GetOutputFormats()
+  {
+    if (_commandLineOptions.TryGetOptionArgumentList(
+      CoverletOptionNames.Formats,
+      out string[]? formats))
+    {
+      LogOptionValue(CoverletOptionNames.Formats, formats, isExplicit: true);
+      return formats[0].Split(',')
+        .Select(f => f.Trim())
+        .ToArray();
+    }
+
+    string[] defaultFormats = new[] { "json", "cobertura" };
+    LogOptionValue(CoverletOptionNames.Formats, defaultFormats, isExplicit: false);
+    return defaultFormats;
+  }
+
+  public string GetOutputSourceMappingDirectory()
```

Comment:
I didn't fully read. But you should call `GetTestResultDirectory` from MTP to determine the directory where the coverage files will be written.

Reply:
coverlet uses a sourceMappingFile for SourceLink. I am not sure whether I keep this option or drop it for coverlet.MTP.

Reply:
And for the directory where the coverage file will be placed? Does it respect `--results-directory` correctly?

Reply:
I removed this option. This additional location for sourceMappingFile is only required for coverlet.console.